src/common/auth.ts

```
@@ -0,0 +1,42 @@
+import { compare, genSalt, hash } from "bcrypt";
+import { psqlClient } from "./database";
+import { sendError, ErrorCodes } from "./error";
+import { User } from "./types";
+import { randomBytes } from "crypto";
+import { salt } from "./constants";
+
+export async function checkCredentials(
+    email: string,
+    password: string
+): Promise<{
+    success: boolean;
+    id?: bigint;
+    error?: "invalidCredentials" | "invalidTOTP" | "accountNotActivated";
+}> {
+    const potentialUser = (await psqlClient.query("SELECT * FROM users WHERE email=$1", [email]))
+        .rows[0] as User;
+    const auth = await compare(password, potentialUser.password);
```

Before doing this you should check if an account with the provided email exists and fail early